scene/resources/gradient.h

```diff
@@ -161,10 +157,10 @@ class Gradient : public Resource {
 				const Point &pointP3 = points[p3];
 
 				float x = (p_offset - pointFirst.offset) / (pointSecond.offset - pointFirst.offset);
-				float r = cubic_interpolate(pointP0.color.r, pointFirst.color.r, pointSecond.color.r, pointP3.color.r, x);
-				float g = cubic_interpolate(pointP0.color.g, pointFirst.color.g, pointSecond.color.g, pointP3.color.g, x);
-				float b = cubic_interpolate(pointP0.color.b, pointFirst.color.b, pointSecond.color.b, pointP3.color.b, x);
-				float a = cubic_interpolate(pointP0.color.a, pointFirst.color.a, pointSecond.color.a, pointP3.color.a, x);
+				float r = Math::cubic_interpolate(pointP0.color.r, pointFirst.color.r, pointSecond.color.r, pointP3.color.r, x);
+				float g = Math::cubic_interpolate(pointP0.color.g, pointFirst.color.g, pointSecond.color.g, pointP3.color.g, x);
+				float b = Math::cubic_interpolate(pointP0.color.b, pointFirst.color.b, pointSecond.color.b, pointP3.color.b, x);
+				float a = Math::cubic_interpolate(pointP0.color.a, pointFirst.color.a, pointSecond.color.a, pointP3.color.a, x);
```

Comment:
This change broke Gradient's cubic interpolation mode. The first and last points use a solid color and the middle points have some kind of an off-by-one error. I tried to fix this myself but only made it worse :/

![image](https://user-images.githubusercontent.com/67974470/179372293-81218b37-7639-496a-a6ed-c9e706d053ef.png)

If it helps, I took the `cubic_interpolate()` method in `Gradient.h` that was removed in this commit and compared it to `Math::cubic_interpolate()` to pinpoint the difference. Since the arguments have different names in both functions, I replaced the first 4 with `A` `B` `C` and `D` and the fifth with `weight`. Comparing them this way made me see that the functions are not the same.

```cpp
Math::cubic_interpolate()
0.5 * ((A * 2.0) + (-C + B) * weight + (2.0 * C - 5.0 * A + 4.0 * B - D) * (weight * weight) + (-C + 3.0 * A - 3.0 * B + D) * (weight * weight * weight));

Gradient.h cubic_interpolate()
B + 0.5 * weight * (C - A + weight * (2.0 * A - 5.0 * B + 4.0 * C - D + weight * (3.0 * (B - C) + D - A)));
```

Reply:
For reference, that bug is being tracked in https://github.com/godotengine/godot/issues/63000.

Reply:
Thanks for finding the exact spot which introduced the regression:)
Regarding your comparison: If I remember correctly I used the simplified version which is only valid for the interval [0,1] (fine for gradients), so it makes sense that the one which was replaced is a bit shorter.